### DISCUSSION TITLE
blog/angular-trpc

### DISCUSSION THREAD

**Discussion by giscus** [angular-experts-io/angular-experts-website-comments#12] [General]
# blog/angular-trpc

Maximum type safety across the entire stack. How to setup a fullstack app with Angular and tRPC.

https://angularexperts.io/blog/angular-trpc

**Comment by jbandi** [comment]
Great article! tRCP seems to become an overarching hype in many frontend communities ...

It's a bit off-topic: How would you go into production with tRCP?

- I would assume any cloud that provides integration with running express/fastify would support tRCP out of the box?
- If you go with Docker would you just run the fastify server in a node-based Docker image?

Do you have any hints or experiences concerning putting tRCP in production?

**Comment by garygrossgarten** [comment]
I'd suggest using defer instead of fromPromise to wrap the trpc client calls. This way, the request to the API will be made once we actually subscribe to the Observable that is being created which is also what the Observables returned by the HttpClient would do.

Example (hint: look at the logs):
https://stackblitz.com/edit/angular-shsahb?file=src%2Fmain.ts

**Comment by BeGj** [comment]
Hi, could you possibly do an updatee on this article from 2023? Considering there has been a lot of improvements over the years on angular ssr.
